### ISSUE TITLE
Wiki Pages can't be sorted by the pageviews

### ISSUE THREAD

**Issue by SidharthBansal** [publiclab/plots2#1928]
### Please describe the problem (or idea)
Wiki pages do not get sorted in the descending order on https://publiclab.org/wiki/popular
![image](https://user-images.githubusercontent.com/20972099/34686475-ae584efa-f4d1-11e7-8e9e-56549d859374.png)

**Comment by SidharthBansal** [comment]
I think it is due to the .order('views DESC') instead of .order('node.views DESC')
https://github.com/publiclab/plots2/blob/646deeb659cd62064792b4134155717d08a5b5a5/app/controllers/wiki_controller.rb#L309-L318
@jwarren Shall I make a pr for it?

**Comment by jywarren** [comment]
Sure, can you confirm your fix in local testing? Thank you!

**Comment by SidharthBansal** [comment]
I was wrong, it was not due to "node." thingy. 
I checked the views and changed the totalviews to views and finally it worked
![image](https://user-images.githubusercontent.com/20972099/34728836-ff4c24d0-f580-11e7-82e1-bda1296fd9e4.png)

**Comment by SidharthBansal** [comment]
Is it due to the issue #1196 ?I guess the error is in the "totalviews". I changed the value of the "views" from the console but the /wiki/popular shows no added number of "pageviews". I tried to change "totalviews" to "views" in the app/views/wiki/_wikis and then it was showing output according to the values of "views" set by the rails console

**Comment by jywarren** [comment]
Hmm, I'm not sure. Can you look for where we increment the view count? It would be under either the wiki controller or potentially a model method in node.rb -- maybe we can diagnose that way. Thanks!!

**Comment by SidharthBansal** [comment]
Yeah under node.rb, total views is given and it points to issue#1196. Impressionist gem is not working properly since then.

**Comment by jywarren** [comment]
Hmm, but it looks like totalviews is actually incrementing -- visit any recent post and there are counted views -- maybe that comment is out of date? I think impressionist must be working. But `totalviews` is from another table, since that's how Impressionist works. I think we have to look on stack overflow or under the Impressionist docs for how to sort by an impressionist-tracked view count?

I updated #1196 to specify that the broken part is just the caching of the tally of views, not that the tallying is not happening at all.  Hope that helps!

**Comment by SidharthBansal** [comment]
Thanks, your hints always works :smile: 
![image](https://user-images.githubusercontent.com/20972099/34913183-28db0b2e-f91c-11e7-911b-97e8ff31cf5e.png)
It is working now.
The error was in the controller we are sorting by node.views and the view we are displaying the node. totalviews where totalviews is a node model method.
So, I just changed the controller to sort according to the totalviews model method by sort_by function.
Order cannot be used as we dont have any totalviews column in any database table.
Just attaching the pr in a few minutes
Thanks
